other/parse_nested_brackets.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @function parseNestedBrackets
+ * @description Parse nested brackets algorithm (recursive implementation) for a string.
+ * @param {string} w - string to parse
+ * @param { openBrackets: string; closingBrackets: string } brackets - object containing the open and closing brackets
+ * @returns {string[]} - array of the tags
+ * @example parseNestedBrackets(`<MAIN hoge><MAIN2 fuga>`) => [ '<MAIN hoge>', '<MAIN2 fuga>' ]
+ * @example parseNestedBrackets(
+ *  `THIS IS SAMPLE TEXT(MAIN hoge 0.1 fuga(ITEM fuga hoge)hoge(ITEM2 nogami(ABBR)))`,
+ *  { openBrackets: '(', closingBrackets: ')' }) => 
+ *  [
+      '(MAIN hoge 0.1 fuga(ITEM fuga hoge)hoge(ITEM2 nogami(ABBR)))',
```

Comment:
How is this recursive? I'd expect the *nested* brackets to be parsed as well.

Reply:
@appgurueu I removed the notation. Could you check this again?